### PR TITLE
feat: upsert devices

### PR DESCRIPTION
**Pull request by Juiced66**
Add an upsert method to devices api

### CODE REVIEW

File: lib/modules/device/DevicesController.ts [hunk 1]
```diff
@@ -56,6 +57,9 @@ export class DevicesController {
             { path: "device-manager/:engineId/devices/:_id", verb: "put" },
           ],
         },
+        upsert: {
+          handler: this.upsert.bind(this),
+        },
```
Comment (afondard) [inline lib/modules/device/DevicesController.ts:62]: It looks like it is missing the http configuration

**Review by afondard — CHANGES_REQUESTED** [review]
Adding the documentation for the upsert may be a good idea?

File: lib/modules/device/DevicesController.ts (hunk 1 above)
Comment (Juiced66) [reply lib/modules/device/DevicesController.ts:62]: oupsyy ty

Comment (Juiced66) [reply lib/modules/device/DevicesController.ts:65]: done

**Review by fmauNeko — APPROVED** [review]

**Review by afondard — APPROVED** [review]

**Review by etrousset — APPROVED** [review]

**Pull request merged**